great_expectations/core/factory/factory.py

```diff
@@ -0,0 +1,22 @@
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar
+
+T = TypeVar("T")
+
+
+class Factory(ABC, Generic[T]):
```

Comment:
I'm not strongly opinionated on this, but is this class YAGNI?

Given that it's basically just an interface at this point, the main use cases would be either a) if we had class/methods that took in Factories without knowing the concrete implementation, or b) to help hold ourselves to the same interface when implementing new Factories (assuming we decide to opt in to inheriting from Factory).

I think my biggest (though still minor) concern is that we define an interface here and then still end up needing workarounds for things, like how our interface for stores didn't hold up so well.

Reply:
Hmm yeah I was torn or whether or not to use it but I added it as a way to hopefully disincentivize bloating these factories and ensuring consistency between them.

If we need workarounds maybe we can revisit? I can go either way here